src/nvim/eval.c

```diff
@@ -13358,26 +13358,31 @@ static void get_maparg(typval_T *argvars, typval_T *rettv, int exact)
     rettv_dict_alloc(rettv);
     if (rhs != NULL) {
       // Return a dictionary.
-      char_u *lhs = str2special_save(mp->m_keys, true);
-      char *const mapmode = map_mode_to_chars(mp->m_mode);
       dict_T *dict = rettv->vval.v_dict;
-
-      dict_add_nr_str(dict, "lhs",     0L, lhs);
-      dict_add_nr_str(dict, "rhs",     0L, mp->m_orig_str);
-      dict_add_nr_str(dict, "noremap", mp->m_noremap ? 1L : 0L, NULL);
-      dict_add_nr_str(dict, "expr",    mp->m_expr    ? 1L : 0L, NULL);
-      dict_add_nr_str(dict, "silent",  mp->m_silent  ? 1L : 0L, NULL);
-      dict_add_nr_str(dict, "sid",     (long)mp->m_script_ID, NULL);
-      dict_add_nr_str(dict, "buffer",  (long)buffer_local, NULL);
-      dict_add_nr_str(dict, "nowait",  mp->m_nowait  ? 1L : 0L, NULL);
-      dict_add_nr_str(dict, "mode",    0L, (char_u *)mapmode);
-
-      xfree(lhs);
-      xfree(mapmode);
+      mapblock_fill_dict(dict, mp, buffer_local);
     }
   }
 }
 
+void mapblock_fill_dict(dict_T *dict, mapblock_T *mp, int buffer_local)
+{
+  char_u *lhs = str2special_save(mp->m_keys, true);
+  char *const mapmode = map_mode_to_chars(mp->m_mode);
+
+  dict_add_nr_str(dict, "lhs",     0L, lhs);
+  dict_add_nr_str(dict, "rhs",     0L, mp->m_orig_str);
+  dict_add_nr_str(dict, "noremap", mp->m_noremap ? 1L : 0L, NULL);
+  dict_add_nr_str(dict, "expr",    mp->m_expr    ? 1L : 0L, NULL);
+  dict_add_nr_str(dict, "silent",  mp->m_silent  ? 1L : 0L, NULL);
+  dict_add_nr_str(dict, "sid",     (long)mp->m_script_ID, NULL);
+  dict_add_nr_str(dict, "buffer",  (long)buffer_local, NULL);
```

Comment:
Note that this would be backwards-incompatible. Fine when checking `if mp.buffer`, not fine when checking actual value. Help explicitly states that `"buffer"` is 1 for a buffer local mapping.